Source/Scene/Label.js

```diff
@@ -1196,5 +1249,184 @@ define([
         return false;
     };
 
+    function declareTypes() {
+        var TextTypes = {
+            LTR : 0,
+            RTL : 1,
+            WEAK : 2,
+            BRACKETS : 3
+        };
+        return freezeObject(TextTypes);
+    }
+
+    function convertTextToTypes(text, rtlDir, rtlChars) {
+        var ltrChars = /[a-zA-Z0-9]/;
+        var bracketsChars = /[()[\]{}<>]/;
+        var parsedText = [];
+        var word = '';
+        var types = declareTypes();
+        var lastType = rtlDir ? types.RTL : types.LTR;
+        var currentType = '';
+        var textLength = text.length;
+        for (var textIndex = 0; textIndex < textLength; ++textIndex) {
+            var character = text.charAt(textIndex);
+            if (rtlChars.test(character)) {
+                currentType = types.RTL;
+            }
+            else if (ltrChars.test(character)) {
+                currentType = types.LTR;
+            }
+            else if (bracketsChars.test(character)) {
+                currentType = types.BRACKETS;
+            }
+            else {
+                currentType = types.WEAK;
+            }
+
+            if (textIndex === 0) {
+                lastType = currentType;
+            }
+
+            if (lastType === currentType && currentType !== types.BRACKETS) {
+                word += character;
+            }
+            else {
+                if (word !== '') {
+                    parsedText.push({Type : lastType, Word : word});
+                }
+                lastType = currentType;
+                word = character;
+            }
+        }
+        parsedText.push({Type : currentType, Word : word});
+        return parsedText;
+    }
+
+    function reverseWord(word) {
+        return word.split('').reverse().join('');
+    }
+
+    function spliceWord(result, pointer, word) {
+        return result.slice(0, pointer) + word + result.slice(pointer);
+    }
+
+    function reverseBrackets(bracket) {
+        switch(bracket) {
+            case '(':
+                return ')';
+            case ')':
+                return '(';
+            case '[':
+                return ']';
+            case ']':
+                return '[';
+            case '{':
+                return '}';
+            case '}':
+                return '{';
+            case '<':
+                return '>';
+            case '>':
+                return '<';
+        }
+    }
+
+    /**
+     *
+     * @param {String} value the text to parse and reorder
+     * @returns {String} the text as rightToLeft direction
+     * @private
+     */
+    function reverseRtl(value) {
+        var rtlChars = /[\u05D0-\u05EA]/;
```

Comment:
So if I'm understanding this code correctly, this regex detects whether the letter is RTL (in this case specifically Hebrew) but otherwise the rest of the code is the same for other languages?  So if we update this regex for other known RTL character sets (perhaps thise discussed [here](https://stackoverflow.com/questions/12006095/javascript-how-to-check-if-character-is-rtl) and [here](https://stackoverflow.com/questions/4330951/how-to-detect-whether-a-character-belongs-to-a-right-to-left-language)  Is that correct?  If so that's awesome and we can easily update this to handle the other languages.

Reply:
> but otherwise the rest of the code is the same for other languages? 

yes. it only necessary when writing a RTL language in Cesium. 

> we can easily update this to handle the other languages.

basically yes, I do not do that, because i cannot read other RTL languages.

Reply:
Cool, we can merge this with just Hebrew and then I'll open a second PR with an updated regex to support additional languages.